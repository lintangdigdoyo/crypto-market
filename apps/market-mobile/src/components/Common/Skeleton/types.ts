export type SkeletonType = "text" | "circular" | "rectangular";
