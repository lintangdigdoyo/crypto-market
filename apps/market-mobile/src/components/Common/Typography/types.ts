export type TypographyVariantType = "title" | "text" | "caption";
