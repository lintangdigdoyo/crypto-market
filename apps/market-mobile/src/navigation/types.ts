import {
  CompositeScreenProps,
  NavigatorScreenParams,
} from "@react-navigation/native";
import { BottomTabScreenProps } from "@react-navigation/bottom-tabs";
import { NativeStackScreenProps } from "@react-navigation/native-stack";

export type RootStackParamList = {
  Root: NavigatorScreenParams<RootTabParamList> | undefined;
};

export type RootStackScreenProps<Screen extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, Screen>;

export type RootTabParamList = {
  TabMarket: undefined;
};

export type RootTabScreenProps<Screen extends keyof RootTabParamList> =
  CompositeScreenProps<
    BottomTabScreenProps<RootTabParamList, Screen>,
    NativeStackScreenProps<RootStackParamList>
  >;
