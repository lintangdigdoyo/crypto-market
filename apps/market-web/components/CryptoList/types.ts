export type PriceTimeType = "day" | "week" | "month" | "year";
