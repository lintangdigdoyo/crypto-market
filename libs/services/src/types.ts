export enum QueryKeyEnum {
  Currencies = "Currencies",
  PriceChanges = "PriceChanges",
}

/** Currencies Types */

export type SupportedCurrenciesResType = {
  code: string;
  message: string;
  payload: SupportedCurrenciesPayloadType[];
};

export type SupportedCurrenciesPayloadType = {
  currencyGroup: string;
  color: string;
  currencySymbol: string;
  name: string;
  logo: string;
  decimal_point: number;
  listingDate: string;
  wallets: WalletType[];
};

export type WalletType = {
  currencyGroup: string;
  tokenSymbol: string;
  decimal_point: number;
  tokenType: string;
  blockchain: string;
  explorer: string;
  listingDate: string;
  blockchainName: string;
  logo: string;
};

/** End of Currencies Types */

/** Price Changes Types */

export type PriceChangesResType = {
  code: string;
  message: string;
  payload: PriceChangePayloadType[];
};

export type PriceChangePayloadType = {
  pair: string;
  latestPrice: string;
  day: string;
  week: string;
  month: string;
  year: string;
};

/** End of Price Changes Types */

export type CurrenciesPriceType = SupportedCurrenciesPayloadType &
  Partial<PriceChangePayloadType>;
