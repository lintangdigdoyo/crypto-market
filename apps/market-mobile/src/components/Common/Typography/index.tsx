import { Text, TextProps, TextStyle } from "react-native";

import { theme } from "../../../styles/palette";
import { TypographyVariantType } from "./types";

interface TypographyProps extends TextProps {
  variant?: TypographyVariantType;
  color?: string;
  weight?: TextStyle["fontWeight"];
  style?: TextStyle;
  children: React.ReactNode;
}

const Typography = ({
  variant = "text",
  color,
  children,
  weight,
  style,
  ...restProps
}: TypographyProps) => {
  return (
    <Text
      style={[
        theme.textVariants[variant],
        weight && { fontWeight: weight },
        !!color && { color },
        !!style && { ...style },
      ]}
      {...restProps}
    >
      {children}
    </Text>
  );
};

export default Typography;
