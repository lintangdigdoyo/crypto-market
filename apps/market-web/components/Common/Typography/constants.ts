export const CLASS_STYLES = {
  heading1: "text-2xl font-bold",
  heading2: "text-xl font-bold",
  paragraph1: "text-lg",
  paragraph2: "text-base",
  caption: "text-sm",
};
