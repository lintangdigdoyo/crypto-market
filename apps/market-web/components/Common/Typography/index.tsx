import { createElement } from "react";
import classNames from "classnames";

import { TypographyType, TypographyVariantType } from "./types";

interface TypographyProps
  extends React.HTMLAttributes<
    HTMLHeadingElement | HTMLParagraphElement | HTMLSpanElement
  > {
  type?: TypographyType;
  variant?: TypographyVariantType;
  className?: string;
  bold?: boolean;
  children: React.ReactNode;
}

const CLASS_STYLES = {
  heading1: "text-2xl font-bold",
  heading2: "text-xl font-bold",
  paragraph1: "text-lg",
  paragraph2: "text-base",
  caption: "text-sm",
};

const Typography = ({
  type = "p",
  variant = "paragraph2",
  className,
  bold,
  children,
  ...restProps
}: TypographyProps) => {
  const Element = createElement(
    type,
    {
      className: classNames(
        /** base styles */
        "base-text",
        /** props styles */
        CLASS_STYLES[variant],
        { ["font-bold"]: bold },
        /** custom styles */
        className
      ),
      ...restProps,
    },
    children
  );
  return <>{Element}</>;
};

export default Typography;
