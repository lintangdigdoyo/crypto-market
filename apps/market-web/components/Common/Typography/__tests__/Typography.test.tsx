import { screen, render, AllByRole } from "@testing-library/react";
import "@testing-library/jest-dom";

import { TypographyType, TypographyVariantType } from "../types";
import Typography from "..";

describe("Typography component", () => {
  it("should renders typography with typography-heading1 class", () => {
    const props = {
      children: <>children</>,
      variant: "heading1" as TypographyVariantType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography).toHaveClass("typography-heading1");
  });

  it("should renders typography with typography-heading2 class", () => {
    const props = {
      children: <>children</>,
      variant: "heading2" as TypographyVariantType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography).toHaveClass("typography-heading2");
  });

  it("should renders typography with typography-paragraph1 class", () => {
    const props = {
      children: <>children</>,
      variant: "paragraph1" as TypographyVariantType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography).toHaveClass("typography-paragraph1");
  });

  it("should renders typography with typography-paragraph2 class", () => {
    const props = {
      children: <>children</>,
      variant: "paragraph2" as TypographyVariantType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography).toHaveClass("typography-paragraph2");
  });

  it("should renders typography with typography-caption class", () => {
    const props = {
      children: <>children</>,
      variant: "caption" as TypographyVariantType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography).toHaveClass("typography-caption");
  });

  it("should renders typography with html P element", () => {
    const props = {
      children: <>children</>,
      type: "p" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("P");
  });

  it("should renders typography with html H1 element", () => {
    const props = {
      children: <>children</>,
      type: "h1" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H1");
  });

  it("should renders typography with html H2 element", () => {
    const props = {
      children: <>children</>,
      type: "h2" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H2");
  });

  it("should renders typography with html H3 element", () => {
    const props = {
      children: <>children</>,
      type: "h3" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H3");
  });

  it("should renders typography with html H4 element", () => {
    const props = {
      children: <>children</>,
      type: "h4" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H4");
  });

  it("should renders typography with html H5 element", () => {
    const props = {
      children: <>children</>,
      type: "h5" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H5");
  });

  it("should renders typography with html H6 element", () => {
    const props = {
      children: <>children</>,
      type: "h6" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("H6");
  });

  it("should renders typography with html SPAN element", () => {
    const props = {
      children: <>children</>,
      type: "span" as TypographyType,
    };
    render(<Typography {...props} />);
    const typography = screen.getByTestId("typography");
    expect(typography.tagName).toBe("SPAN");
  });
});
