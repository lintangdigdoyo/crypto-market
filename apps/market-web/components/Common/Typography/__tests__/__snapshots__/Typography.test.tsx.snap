// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should renders typography correctly 1`] = `
<p
  className="typography-base typography-paragraph2"
  data-testid="typography"
>
  children
</p>
`;
